app/src/main/kotlin/compiler/intermediate_form/FunctionDependenciesAnalyzer.kt

```diff
@@ -24,18 +24,39 @@ object FunctionDependenciesAnalyzer {
             return uniqueIdentifiers[function]!!.value
         }
 
-        fun analyze(node: Any, pathSoFar: String?) {
+        fun analyze(node: Any, pathSoFar: String? = null) {
             when (node) {
                 is Program.Global.FunctionDefinition -> { analyze(node.function, pathSoFar) }
                 is Statement.FunctionDefinition -> { analyze(node.function, pathSoFar) }
                 is Function -> {
                     val newPrefix = nameFunction(node, pathSoFar)
-                    node.body.forEach { analyze(it, newPrefix) }
+                    var blockNumber = -1
```

Comment:
What if I have 2 nested functions `functionName`, and a nested function `functionName1`. Seems to be a problem. What do you think?

Reply:
You're right, this could cause problems. I added the `@block` prefix before the block number. This will never appear in a regular function name - the character `@` is not allowed.